docs/proposal-0004-jetstream-preview.md

```diff
@@ -0,0 +1,118 @@
+--sql-output-dir          Path to write executed SQL 
+--platform                Platform/app to run analysis for
+```
+
+By default the preview would run jetstream on a 3 day analysis window on a population sample of 1%.
+These default parameters can be overwritten when invoking `jetstream preview`.
+
+The resulting data artifacts get written into the `mozdata.tmp` dataset by default, which is configured to delete data after 7 days. Anyone at Mozilla has permissions to write to this dataset.
+
+Once jetstream has generated the preview data it will print out a URL to a Looker dashboard. The Looker dashboard allows users to visualize the preview data. 
+
+The dashboard is a static dashboard that has been added to Looker. It has a filter field that will be pre-filled throught the URL pointing to the temporary datasets that have been generated. It will also have filters so users can select the analysis period, analysis basis, segment, metric and statistic they would like to preview. The visualization will be a line graph with confidence intervals that shows a line for every branch and comparison.
+
+The queries that get executed will be saved locally into files which makes it possible for users to debug the queries (enrollment and metrics) that get executed.
+
+Running the preview command will also print out cost estimates for running analyses on the experiment.
+
+The jetstream CI will need to be updated to publish jetstream to pypi. This way users do not need to pull in the jetstream repository, but can simply run `pip install mozilla-jetstream`.
+
+Jetstream fetches information on how experiments are configured from the Experimenter API. In some cases users might want to get a preview for experiments that haven't been launched yet or that haven't seen any enrollments. For these experiments a random population sample can be generated when specifying `--generate-population` that will be considered as the clients that enrolled.
+
+### Other ideas
+
+Metadata around cost could be stored in a temporary BigQuery table and displayed on the preview dashboard. It might be possible to also store and display the queries that were run as part of the preview including errors that occured. This would help with debugging queries early on.
+
+## Experimenter API
+
+Experimenter only has an API that surfaces experiments that are currently live. To run the preview on experiments that are currently in draft mode, changes would need to be made to surface these experiments through the API.
+
+Having access to experiments in preview would also help users to write and validate configs before the experiment has launched. Currently any pull-request adding a config for an experiment that isn't live cannot be merged or even be validated.
+
+## Risks
+
+There are some potential risks with providing users with the option of generating previews:
+* Users might optimize their configs based on the preview data which could impact the experiment results
```

Comment:
What do we mean by "optimize" here? Like, they add/remove segments until they see the most statistically significant results? That would be a bad situation, but I'm not sure the best way to prevent that other than by education & code review (if they do this through the CI, someone will see the log and could comment). Right now, nothing prevents DSes from doing something like this manually

Reply:
Yeah, that's what I meant. I guess we can't really prevent this, even in the current state.

Reply:
Is there a scenario where the preview generates results that look wrong but are actually valid that might cause a change to a datasource query or something? That's what I was thinking of when I read this -- less intentionally hacking, more "fixing" things that look broken but are actually just surprising. I'm not sure if the `preview` results would even give enough info for this kind of scenario though.